src/plugins/intel_gpu/tests/unit/test_cases/select_gpu_test.cpp

```diff
@@ -71,6 +71,85 @@ TEST(select_gpu_f32, select_basic) {
     test_select_basic<float>(false);
 }
 
+template <typename T>
+void test_select_basic_5d(bool is_caching_test) {
+    auto& engine = get_test_engine();
+
+    auto input = engine.allocate_memory({ data_types::f32, format::bfzyx, { 2, 2, 2, 2, 2 } });
+    auto input2 = engine.allocate_memory({ data_types::f32, format::bfzyx, { 2, 2, 2, 2, 2 } });
+    auto mask = engine.allocate_memory({ data_types::f32, format::bfzyx, { 2, 2, 2, 2, 2 } });
+
+    topology topology;
+    topology.add(input_layout("input", input->get_layout()));
+    topology.add(input_layout("input2", input2->get_layout()));
+    topology.add(input_layout("mask", mask->get_layout()));
+    topology.add(cldnn::select("select", input_info("mask"), input_info("input"), input_info("input2")));
+
+    set_values(input, {
+        1.f,  0.f,  5.f,  1.5f,
+        2.f,  0.f,  6.f,  5.2f,
+        3.f,  0.5f, 7.f, 12.f,
+        4.f, -0.5f, 8.f,  8.f,
+        1.f,  0.f,  5.f,  1.5f,
+        2.f,  0.f,  6.f,  5.2f,
+        3.f,  0.5f, 7.f, 12.f,
+        4.f, -0.5f, 8.f,  8.f
+    });
+
+    set_values(input2, {
+         0.5f, 2.5f,  0.5f,  2.5f,
+         5.f,  7.f,   2.f,   4.f,
+        15.f, 17.f,   8.f,  10.f,
+        -2.f,  6.5f, -0.5f, -2.5f,
+         0.5f, 2.5f,  0.5f,  2.5f,
+         5.f,  7.f,   2.f,   4.f,
+        15.f, 17.f,   8.f,  10.f,
+        -2.f,  6.5f, -0.5f, -2.5f
+    });
+
+    set_values(mask, {
+        0.f, 0.f, 0.f, 0.f,
+        1.f, 1.f, 1.f, 1.f,
+        0.f, 1.f, 0.f, 1.f,
+        1.f, 0.f, 1.f, 0.f,
+        0.f, 0.f, 0.f, 0.f,
+        1.f, 1.f, 1.f, 1.f,
+        0.f, 1.f, 0.f, 1.f,
+        1.f, 0.f, 1.f, 0.f
+    });
+
+    cldnn::network::ptr network = get_network(engine, topology, get_test_default_config(engine), get_test_stream_ptr(), is_caching_test);
+
+    network->set_input_data("input", input);
+    network->set_input_data("input2", input2);
+    network->set_input_data("mask", mask);
+    auto outputs = network->execute();
+
+    auto output = outputs.at("select").get_memory();
+
+    float answers[32] = {
+         0.5f, 2.5f, 0.5f, 2.5f,
+         2.f,  0.f,  6.f,  5.2f,
+        15.f,  0.5f, 8.f, 12.f,
+         4.f,  6.5f, 8.f, -2.5f,
+         0.5f, 2.5f, 0.5f, 2.5f,
+         2.f,  0.f,  6.f,  5.2f,
+        15.f,  0.5f, 8.f, 12.f,
+         4.f,  6.5f, 8.f, -2.5f
+    };
+
+    cldnn::mem_lock<T> output_ptr(output, get_test_stream());
+
+    for (int i = 0; i < 32; i++)
+    {
+        ASSERT_TRUE(are_equal(answers[i], output_ptr[i]));
```

Comment:
What about using `ASSERT_EQ` or `ASSERT_NEAR`?

Reply:
Updated as you commented.